src/hw1/vector.py

```diff
@@ -0,0 +1,71 @@
+import math
+from src.util.type_checking import is_number
+
+
+class Vector:
+    def __init__(self, first, *args):
+        if not all(is_number(element) for element in args):
+            raise TypeError("A vector can't have a non-numeric element!")
+
+        self.elements = [first] + list(args)
+        self.length = 1 + len(args)
```

Comment:
This looks a bit awkward. Why do we need a specific "first"? Why not do just args, but add a check that it's not empty?